### ISSUE TITLE
Miyagi working?

### ISSUE THREAD

**Issue by MythicalXFaze** [Sniffleupagus/pwnagotchi_plugins#5]
Hey snuff, i see you're still active which makes me very happy... i was looking for a way to contact you and i know this is not the best way. but i am not understanding what miyagi is going. I have my pwnagotchi and it's been running for 2 hours and 20 minutes. Laziness has been set to 94%+ according to the UI. the mode section updates with TR01 but has not changed. i know that its training because of that but im just not understanding what its doing or if the lazyness should be that high. Plus i have no idea how long an epoch is but i was told it's like a billion years. I'm just looking for clarification on what it's doing. 

Is it a problem if i let it run on its own without introducing it to other networks. will it still function correctly while in training mode?

ALSO one thing i would like help with pertaining to the code. how to i mode the Train:
    Lazy:

to the corner of my screen instead of clipping in the face and mouth. 

Ps. I have 0 coding background and i'm extremely new to all this and i'm just looking for some help/clarification. Thanks!

**Comment by Sniffleupagus** [comment]
Oh yeah..  I think there's a bug in the main pwnagotchi the way training gets calls. I think it only calls on the first epoch. I believe I modified my pwnagotchi code to call it every epoch.  that might be the problem.  Its been a while, so I could be wrong

**Comment by Sniffleupagus** [comment]
I believe the problem is the recent versions of stable_baselines_3 package are no longer backwards compatible with pwnagotchi. It calls the "starting training" event, but never calls a training step. There are some error messages in the pwnagotchi.log about "render called with no mode/type/something"... it's been a while, but something like that.
The solution I found was to revert stable_baselines_3 to version 1.8.0.  That also requires gym-0.20, which has a bug that breaks newer versions of setup_tools.  There is a patched version of gym 0.20 that works...

TLDR: yes, miyagi is working, but pwnagotchi AI is broken :)

This should fix it. Might need to run as "sudo", might need "--break-system-packages", might need to activate a venv, depending on how it is set up.

pip3 install git+https://github.com/CeyaoZhang/gym.git@v0.21.0
pip3 install stable_baselines_3==1.8.0

**Comment by Sniffleupagus** [comment]
Closing this, as it was a problem in the pwnagotchi install, nothing the plugin can fix.
